pycbc/inference/sampler/cpnest.py

```diff
@@ -0,0 +1,219 @@
+    pool : function with map, Optional
+        A provider of a map function that allows a function call to be run
+        over multiple sets of arguments and possibly maps them to
+        cores/nodes/etc.
+    """
+    name = "cpnest"
+    _io = CPNestFile
+
+    #burn_in_class = MCMCBurnInTests
+
+    def __init__(self, model, nlive, checkpoint_interval=None,
+                 logpost_function=None, nthreads=8, maxmcmc=1000,resume=True):
+
+        self.model = model
+        self.nlive = nlive
+        self.maxmcmc = maxmcmc
+        self.nthreads = nthreads
+        #self.names = model.sampling_params
+        # create a wrapper for calling the model
+        if logpost_function is None:
+            logpost_function = 'logposterior'
+        self.model_call = CPnestModel(model)
+        self._sampler = None
+        #self._samples = None
+        self._nested_samples = None
+        self._posterior_samples = None
+        self._logz = None
+        self._dlogz = None
+        self.checkpoint_file = 'test.out' # !!!!!!!!!FIXME!!!!!!!!!!
+    
+    def run(self, **kwargs):
+        OUTDIR = os.path.dirname(os.path.abspath(self.checkpoint_file))
+        if self._sampler is None:
+            self._sampler = cpnest.CPNest(self.model_call,verbose=1,output=OUTDIR,nthreads=self.nthreads,nlive=self.nlive,maxmcmc=self.maxmcmc,resume=True)
+        res = self._sampler.run()
+
+    @property
+    def io(self):
+        return self._io
+
+    @property
+    def niterations(self):
+        return len(self.samples.values()[0])
+
+    @classmethod
+    def from_config(cls, cp, model,nprocesses=1,use_mpi=False):
+        """
+        Loads the sampler from the given config file.
+        """
+        section = "sampler"
+        # check name
+        assert cp.get(section, "name") == cls.name, (
+            "name in section [sampler] must match mine")
+        # get the number of live points to use
+        nlive = int(cp.get(section, "nlive"))
+        maxmcmc = int(cp.get(section, "maxmcmc"))
+        nthreads = int(cp.get(section, "nthreads"))
+        obj = cls(model, nlive=nlive, maxmcmc=maxmcmc,
+                  nthreads=nthreads)
+        # set target
+        #obj.set_target_from_config(cp, section)
+        # add burn-in if it's specified
+        #obj.set_burn_in_from_config(cp)
+        return obj
+
+    def checkpoint(self):
+        pass
+
+    def finalize(self):
+        logz = self._sampler.NS.logZ
+        dlogz = 0.1 #######FIXME!!!!!###############
+        logging.info("log Z, dlog Z: {}, {}".format(logz, dlogz))
+        for fn in [self.checkpoint_file]:
+            with self.io(fn, "a") as fp:
+                fp.write_logevidence(logz,dlogz)
+        self.write_results(self.checkpoint_file)
+        logging.info("Writing samples to files")
+	for fn in [self.checkpoint_file, self.backup_file]:
+	    self.write_results(fn)
+	    with self.io(fn, "a") as fp:
+	        fp.write_niterations(self.niterations)
```

Comment:
Move  this call to `write_niterations` to `write_results`, so you don't have to reopen the file here.